internal/engine/upper_test.go

```diff
@@ -2030,6 +2069,31 @@ func (f *testFixture) WriteConfigFiles(args ...string) {
 	f.store.Dispatch(manifestFilesChangedAction{manifestName: ConfigsManifestName, files: filenames})
 }
 
+func (f *testFixture) setupHAProxy() model.Manifest {
```

Comment:
consider renaming this method to be obvious that we're setting up "some docker-compose test junk" instead of like, an actual proxy doing actual things.

also, instead of tracking origWD and reading then writing files, you could just store the file contents strings -- see [testyaml.go](https://github.com/windmilleng/tilt/blob/6dc5d88efeac03ec0cc2da3ed3c26ccb30fb8c41/internal/k8s/testyaml/testyaml.go)

Reply:
I have had a TON of problems with indentations when storing YAML that way so I am trying to sidestep it for now